proxy/proxy_test.go

```diff
@@ -1332,6 +1332,289 @@ var _ = Describe("Proxy", func() {
 			Expect(b[len(b)-1]).To(Equal(byte('\n')))
 		})
 
+		Context("A slow response body", func() {
+			BeforeEach(func() {
+				conf.EndpointTimeout = 5 * time.Second
+			})
+
+			It("shows in response_time, not gorouter_time", func() {
+				ln := test_util.RegisterConnHandler(r, "slow-response-app", func(conn *test_util.HttpConn) {
+					_, err := http.ReadRequest(conn.Reader)
+					Expect(err).NotTo(HaveOccurred())
+
+					body := "this body took two seconds."
+					conn.WriteLines([]string{
+						"HTTP/1.1 200 OK",
+						fmt.Sprintf("Content-Length: %d", len(body)),
+					})
+
+					time.Sleep(2 * time.Second)
+
+					conn.WriteLine(body)
+					conn.Close()
+				})
+				defer ln.Close()
+
+				conn := dialProxy(proxyServer)
+
+				req := test_util.NewRequest("GET", "slow-response-app", "/", nil)
+
+				started := time.Now()
+				conn.WriteRequest(req)
+
+				resp, err := http.ReadResponse(conn.Reader, &http.Request{})
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(resp.StatusCode).To(Equal(http.StatusOK))
+				Expect(time.Since(started)).To(BeNumerically("<", 5*time.Second))
+
+				Eventually(func() (int64, error) {
+					fi, err := f.Stat()
+					if err != nil {
+						return 0, err
+					}
+					return fi.Size(), nil
+				}, "5s").ShouldNot(BeZero())
+
+				b, err := os.ReadFile(f.Name())
+				Expect(err).NotTo(HaveOccurred())
+				Expect(strings.HasPrefix(string(b), "slow-response-app - [")).To(BeTrue())
+				Expect(string(b)).To(ContainSubstring(`response_time:`))
+				Expect(string(b)).To(ContainSubstring(`gorouter_time:`))
+				Expect(string(b)).To(MatchRegexp("response_time:2"))
+				Expect(string(b)).To(MatchRegexp("gorouter_time:0"))
+			})
+
+			Context("A slow app with multiple broken endpoints and attempt details logging enabled", func() {
+				BeforeEach(func() {
+					conf.EndpointDialTimeout = 1 * time.Second
+					conf.Logging.EnableAttemptsDetails = true
+					conf.DropletStaleThreshold = 1
+				})
+
+				It("shows in failed_attempts_time and backend_time, not gorouter_time", func() {
+
+					// Register some broken endpoints to cause retries
+					test_util.RegisterAddr(r, "partially-broken-app", "10.255.255.1:1234", test_util.RegisterConfig{InstanceIndex: "1"})
+					test_util.RegisterAddr(r, "partially-broken-app", "10.255.255.1:1235", test_util.RegisterConfig{InstanceIndex: "2"})
+
+					ln := test_util.RegisterConnHandler(r, "partially-broken-app", func(conn *test_util.HttpConn) {
+						_, err := http.ReadRequest(conn.Reader)
+						Expect(err).NotTo(HaveOccurred())
+
+						body := "this body took two seconds."
```

Comment:
well, it may give you some false sense of security to be able to up this value without consequence - which is false. it also depends on the [EndpointTimeout](https://github.com/cloudfoundry/gorouter/pull/366/files#diff-8d5bed61a3c0b7b0bfdd65dd4c93e647568b1a402840f5e302c0c922f6c58ea3R1337) value. So I don't see too much gain vs. risk here and rather not change it.